tests/PHPStan/Analyser/nsrt/array-intersect-key.php

```diff
@@ -15,9 +15,9 @@ class Foo
 	public function nonEmpty(array $arr, array $arr2): void
 	{
 		assertType('non-empty-array<int|string, string>', array_intersect_key($arr));
-		assertType('non-empty-array<int|string, string>', array_intersect_key($arr, $arr));
+		assertType('array<int|string, string>', array_intersect_key($arr, $arr));
```

Comment:
I'm not sure it's worth adding complexity to the ArrayIntersectKeyFunctionReturnTypeExtension in order to check if all the variable are the same. This array_intersect_key  call is useless...

Reply:
The test is only about the types. No need for checking identical vars passed

Reply:
Then 
```
array_intersect_key($nonEmptyArray1, $nonEmptyArray2)
```
is now correctly understood as an array that may be empty.